testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/script/DeployedScriptMapperTest.java

```diff
@@ -74,20 +72,21 @@ public static JavaArchive deploy() throws IOException {
     @BeforeClass
     public static void verifyEnvironment() {
         ContainerAssume.assumeNotAuthServerUndertow();
-        ContainerAssume.assumeNotAuthServerQuarkus();
     }
 
     @ArquillianResource
     private Deployer deployer;
 
     @Before
-    public void configureFlows() {
+    public void configureFlows() throws Exception {
         deployer.deploy(SCRIPT_DEPLOYMENT_NAME);
+        reconnectAdminClient();
```

Comment:
@pedroigor @vmuzikar I tried to be as minimally invasive as possible w.r.t. the existing testsuite. That lead to a slight "pollution" for the tests with this reconnect. Problem is: We're using `adminClient` from `AbstractKeycloakTest` directly, assuming it is always connected. But it isn't when we restart the server in the test / after deploying. Another possibility would be e.g. to create a `getRestartAgnosticAdminClient` in `AbstractKeycloakTest` (or a supplier, fwiw) and encapsulate the "reconnect if not exists or closed" logic into that and use this for all tests. But I didn't want to go over the whole testsuite (e.g. ApiUtil, etc.) and change all calls which are right now using `adminClient` directly. 
Also, I first added a check for àuth-server-quarkus`, but this would have lead to more loc, and the `reconnectAdminClient` is not that expensive, it just sets the `adminClient` in `AbstractKeycloakTest` when there is no client or the client is closed, so I didn't add. WDYT of this approach?

Reply:
IMO, a better approach would be to re-connect within the `KeycloakQuarkusServerDeployableContainer`. There you should be able to inject the `TestContext` instance (it is application scoped) and, when starting the server, check whether an instance is available (in case of a re-start it should be there) and force a re-connect.

Reply:
As said in our call: There's a problem then because the actual `adminClient` used in the tests (and many more places throughout the testsuite) is not necessarily the one we set via `reconnectAdminClient()` or even `setAdminClient(new..)` in TestContext.

Both, `TestContext` and `AbstractKeycloakTest` have a `reconnectAdminClient`-method, and to be sure the client is really reconnected, one has to call `AbstractKeycloakTest::reconnectAdminClient()`, which is not possible in our container (I first thought about reflection, but that won't fit either without many instanceof calls, because the hierarchy until we reach the AbstractKeycloakTest class is different for the tests).

Core reason is our testsuite setup does not expect the server to restart when deploying something, but that's how Keycloak.X works by design.
 
So with that assumption in place, the adminClient was used all over the place directly. As said before, one could replace all direct usages of `AbstractKeycloakTest::adminClient` with a method / supplier which then encapsulates the "reconnect if closed" logic, but that would be a kind of major refactoring, and I wanted to avoid that for now.  :)